src/main/webapp/app/exercises/file-upload/manage/file-upload-exercise-detail.component.html

```diff
@@ -12,6 +12,53 @@ <h2><span jhiTranslate="artemisApp.fileUploadExercise.detail.title">File Upload
                     [courseId]="fileUploadExercise.course.id!"
                     [exercise]="fileUploadExercise"
                 ></jhi-non-programming-exercise-detail-common-actions>
+                <a
+                    *ngIf="fileUploadExercise.isAtLeastTutor && fileUploadExercise.course"
+                    [routerLink]="['/course-management', fileUploadExercise.course!.id!, 'file-upload-exercises', fileUploadExercise.id!, 'exercise-statistics']"
+                    class="btn btn-info btn-sm mr-1"
+                >
+                    <fa-icon [icon]="'chart-bar'"></fa-icon>&nbsp;<span jhiTranslate="statistics.statistics-title">Statistics</span>
+                </a>
+            </div>
+            <div *ngIf="fileUploadExercise.releaseDate != undefined && moment(fileUploadExercise.releaseDate).isBefore(moment())">
+                <jhi-doughnut-chart
+                    class="mt-3 d-inline-flex"
+                    [courseId]="fileUploadExercise.course?.id!"
+                    [contentType]="DoughnutChartType.AVERAGE_EXERCISE_SCORE"
+                    [exerciseId]="fileUploadExercise.id!"
+                    [exerciseType]="ExerciseType.TEXT"
+                    [currentPercentage]="doughnutStats?.averageScoreOfExercise"
+                    [currentAbsolute]="absoluteAveragePoints"
+                    [currentMax]="doughnutStats?.maxPointsOfExercise"
+                >
+                </jhi-doughnut-chart>
+                <jhi-doughnut-chart
+                    class="mt-3 d-inline-flex"
+                    [courseId]="fileUploadExercise.course?.id!"
+                    [contentType]="DoughnutChartType.PARTICIPATIONS"
+                    [exerciseId]="fileUploadExercise.id!"
+                    [exerciseType]="ExerciseType.TEXT"
+                    [currentPercentage]="participationsInPercent"
+                    [currentAbsolute]="doughnutStats?.numberOfParticipations"
+                    [currentMax]="doughnutStats?.numberOfStudentsInCourse"
+                >
+                </jhi-doughnut-chart>
+                <jhi-doughnut-chart
+                    class="mt-3 d-inline-flex"
+                    [courseId]="fileUploadExercise.course?.id!"
+                    [contentType]="DoughnutChartType.QUESTIONS"
+                    [exerciseId]="fileUploadExercise.id!"
+                    [exerciseType]="ExerciseType.TEXT"
+                    [currentPercentage]="questionsAnsweredInPercent"
+                    [currentAbsolute]="doughnutStats?.numberOfAnsweredQuestions"
+                    [currentMax]="doughnutStats?.numberOfQuestions"
+                >
```

Comment:
This part about 3 doughnut - **jhi-doughnut-chart** s  is repeated many times for different types of exercises. It looks like it could be reused. I'm wondering what you think about making this chunk a modular/common reusable component containing the 3 charts.